server-spi-private/src/main/java/org/keycloak/models/session/PersistentUserSessionAdapter.java

```diff
@@ -268,6 +275,9 @@ public int hashCode() {
         @JsonProperty("ipAddress")
         private String ipAddress;
 
+        @JsonProperty("deviceInfo")
+        private DeviceInfo device;
```

Comment:
For example here. If this is kept inside `notes`, there is no need to change this inner class.